levels/challenge-questions/events.js

```diff
@@ -5,59 +5,117 @@ const fixDoorWallDepthSorting = require("./events/fixDoorDepthSorting");
 
 const WORLD_STATE_KEY = "com.twilioquest.challenge-questions";
 
-const INITIAL_STATE = {};
-
-module.exports = function (event, world) {
-  const worldState = merge(INITIAL_STATE, world.getState(WORLD_STATE_KEY));
-
-  const lockingObjectives = [];
+const INITIAL_STATE = {
+  challengeMapsCompleted: 0,
+  playerHasTeleported: false,
+  playerCompletedObjective: false
+};
 
-  if (event.name === 'mapDidLoad')
-    worldState.currentMapName = event.mapName;
+function mapIsCompleted(world, lockingObjectives, ...ignoredObjectives) {
+  return lockingObjectives && lockingObjectives.length &&
+    lockingObjectives.every((objectiveName) => {
+      if (!ignoredObjectives.includes(objectiveName))
+        return world.isObjectiveCompleted(objectiveName);
+      
+      return true;
+    });
+}
 
-  if (event.name === 'conversationDidEnd' && event.npc.conversation === 'cedricDefault' && worldState.chosenMap) {
-    world.warp('challenge-questions', 'player_entry1', worldState.chosenMap);
-    delete worldState.chosenMap;
-  }
+function getMapEvent(mapName) {
+  const mapEvent = {};
 
-  switch(worldState.currentMapName) {
+  switch(mapName) {
     case 'default':
-      lockingObjectives.push(
-        'difference-max-min',
-        'remove-duplicate-characters',
-        'reverse-words',
-        'sum-array'
-      );
+      mapEvent.lockingObjectives = ['difference-max-min', 'remove-duplicate-characters', 'reverse-words', 'sum-array', 'ducktypium-helm'];
```

Comment:
Do we use `mapEvent.lockingObjectives` anywhere that isn't in the `mapIsCompleted` function? 

Right now, it seems like we're adding the `ducktypium-helm` and `ducktypium-chest` objectives and then filtering them back out during the `mapIsCompleted` check.

If lockingObjectives isn't used anywhere else, we could probably simplify this logic and just remove those objectives from the array. And then get rid of the `ignoredObjectives` logic.